### PR TITLE
feat: add BN254 precompiles

### PR DESCRIPTION
**Pull request by jonathanpwang**

### CODE REVIEW

**Comment by luffykai** [comment]
https://github.com/axiom-crypto/openvm-reth-benchmark/actions/runs/12387173926/job/34576186347 it runs, but need to find a good block to see the diff

File: bin/client-eth/Cargo.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ openvm-client-executor = { path = "../../crates/executor/client" }
 
 revm-precompile = { version = "11.0.1", default-features = false, features = [
     "openvm-k256",
+    # "openvm-bn",
```
Comment (luffykai) [inline bin/client-eth/Cargo.toml:15]: uncomment

**Comment by yi-sun** [comment]
You can try 21_000_000 to 21_000_003 since we already have benchmarks for those from before.

**Comment by luffykai** [comment]
turned off bn precompile: https://github.com/axiom-crypto/openvm-reth-benchmark/blob/gh-pages/benchmarks-dispatch/refs/heads/feat/bn-precompiles/reth-dd176307c4dc3b0d8d946cf0c81776a7a8613ba0e2298d3c771544264b0cb968.md
this run https://github.com/axiom-crypto/openvm-reth-benchmark/actions/runs/12399814599/job/34617109887

**Comment by luffykai** [comment]
with bn precompile: https://github.com/axiom-crypto/openvm-reth-benchmark/actions/runs/12401490882

just triggered, will update

**Comment by jonathanpwang** [comment]
the diff: https://github.com/axiom-crypto/openvm-reth-benchmark/commit/1c3d64c7e588fe9a3a2a8cfcd8fc303ceed9c15c

**Review by luffykai — APPROVED** [review]

**Pull request merged**